### ISSUE TITLE
sbt 1.4.x run task does not pass EOF condition on to application

### ISSUE THREAD

**Issue by klaeufer** [sbt/sbt#5999]
## steps

```
$ sbt run
[info] running edu.luc.cs.laufer.cs473.expressions.CombinatorCalculator 
Enter infix expression: 3
You entered: 3
Enter infix expression: 4
You entered: 4
Enter infix expression: Ctrl-D 5
6
7
^C
```

## problem

After entering Ctrl-D, the application invoked through sbt does not terminate (as it correctly does for earlier versions of sbt), and it no longer seems to receive any subsequent input. This suggests that sbt stops passing input to the application but does not exit either.

## expectation

The application invoked through sbt should receive the EOF condition and exit.

## notes

sbt 1.4.0, 1.4.1
Scala 2.13.3
AdoptOpenJDK Java 11.0.8
Ubuntu 18.04

**Comment by eatkins** [comment]
Thanks for the report. Do you have a sample project? I tried https://github.com/LoyolaChicagoCode/expressions-scala and while I did uncover a bug in sbt's handling of ctrl+d (#6000), I was unable to reproduce the specific behavior you reported.

**Comment by eatkins** [comment]
I figured out what was going on. This should be fixed by #6003.

**Comment by klaeufer** [comment]
> Thanks for the report. Do you have a sample project? I tried https://github.com/LoyolaChicagoCode/expressions-scala and while I did uncover a bug in sbt's handling of ctrl+d (#6000), I was unable to reproduce the specific behavior you reported.

Sorry, I was tied up... Yes, that is the sample project I used for this report. I double-checked and was able to reproduce this behavior for sbt 1.4.0 and 1.4.1. But it sounds like you've fixed it by now! Thanks for the prompt action.
